ocaml/xapi/helpers.ml

```diff
@@ -1768,39 +1768,52 @@ end = struct
 
   let systemctl_ cmd = systemctl cmd |> ignore
 
-  let is_enabled () =
-    let is_enabled_stdout =
-      try systemctl "is-enabled"
-      with
-      (* systemctl is-enabled appears to return error code 1 when the service is disabled *)
-      | Forkhelpers.Spawn_internal_error (stderr, stdout, status) as e -> (
-        match status with
-        | Unix.WEXITED n
-          when n = 1
-               && Astring.String.is_prefix ~affix:"disabled" stdout
-               && Astring.String.is_empty stderr ->
-            "disabled"
-        | _ ->
-            raise e
-      )
-    in
-    is_enabled_stdout |> Astring.String.trim |> function
-    | "enabled" ->
-        true
-    | "disabled" ->
-        false
-    | unknown ->
-        D.error
-          "Stunnel.is_enabled: expected 'enabled' or 'disabled', but got: %s"
-          unknown ;
-        false
+  let systemctl_socket cmd = call_script !Xapi_globs.systemctl [cmd; "stunnel-on-demand.socket"]
```

Comment:
Rob HoesYesterday
Why no stunnel-on-demand@xapi here?

Ming Lu （陆明）2 hours ago
Indeed. I added a systemctl stop stunnel-on-demand@*.service to clean it up in restarting.
Otherwise, systemctl disable stunnel-on-demand.socket will clean up all running service instances.